### ISSUE TITLE
Sequence.index should support start and stop arguments in Py >= 3.5

### ISSUE THREAD

**Issue by telendt** [python/typeshed#486]
From [collections.abc.Sequence](https://docs.python.org/3/library/collections.abc.html#collections.abc.Sequence) doc:
> Changed in version 3.5: The index() method added support for stop and start arguments.

Given [`typing.Sequence`](https://docs.python.org/3/library/typing.html#typing.Sequence) is just a generic version of `collections.abc.Sequence` I would expect the following code to have correct type declarations (at least in Python 3.5):

```
from typing import Sequence, TypeVar

T = TypeVar('T')

def index(a: Sequence[T], x: T) -> int:
    try:
        return a.index(x, 0, len(a))
    except ValueError:
        return -1
```

But I get the following:

```
$ mypy --python-version 3.5 test.py
test.py: note: In function "index":
test.py:7: error: Too many arguments for "index" of "Sequence"
```

**Comment by gvanrossum** [comment]
Does such code (using typing.Sequence) work st runtime? If not we need to
fic typing.py as well. If yes, we just need a PR for typing.pyi in typeshed.



-- 
--Guido (mobile)

**Comment by telendt** [comment]
@gvanrossum: The code works at runtime (now, I had a small bug in my example which I've just fixed). 
I can provide PR for `typing.pyi`. The only question I have it whether I should really provide support for these arguments for Python >= 3.5 only?
`start` and `stop` are supported in `index` of all builtin sequences (list, str, bytes...) for a long time, but they were added to Sequence ABC in Python 3.5 ([issue 23086](https://bugs.python.org/issue23086)). Unfortunately it didn't get backported to older Python versions. So in older Python versions one can create custom sequence which would have index without start/stop support...
